x-pack/solutions/observability/plugins/observability_onboarding/server/lib/get_agent_version.ts

```diff
@@ -24,9 +27,32 @@ export async function getAgentVersionInfo(
     agentClient.getLatestAgentAvailableBaseVersion(includeCurrentVersion),
     agentClient.getLatestAgentAvailableDockerImageVersion(includeCurrentVersion),
   ]);
+  const agentTargetVersion =
+    fromVersion && upToVersion
+      ? getLatestAgentVersionInRange(
+          await agentClient.getAvailableVersions(),
+          kibanaVersion,
+          fromVersion,
+          upToVersion
+        )
+      : undefined;
   return {
     agentVersion,
     agentBaseVersion,
     agentDockerImageVersion,
+    agentTargetVersion,
   };
 }
+
+export function getLatestAgentVersionInRange(
+  agentVersionList: string[],
+  kibanaVersion: string,
+  fromVersion: string,
+  upToVersion: string
+): string {
+  return (
+    agentVersionList.find((version) => {
+      return semver.satisfies(version, `>=${fromVersion} <${upToVersion}`);
```

Comment:
I'm no native speaker but I think "up to" means "less than or equal to" so the naming tripped me a bit when I saw the examples in the unit tests. 

I'm trying to think of alternatives to express the range constraint more clearly but it's all quite verbose in English. 

Would two major versions up ever be a valid use case (e.g. `getLatestAgentVersionInRange([], '', '9.1', '11')`) or are we actually saying that we want to ensure we stay within the same major version as the `fromVersion` parameter? If that's the case we don't need the `upToVersion` parameter at all since we can calculate that using the semver package)

Reply:
Ah, my bad, I thought "up to" means "less then", that's an unexpected TIL :) thank you for pointing out! Yeah, this whole change is so cumbersome, it's essentially a workaround patch following a confusing agent versioning. The intent with those constraints was to stay within the same major version, maybe a single parameter like `9.x.x` would work better, I'll try that.

Reply:
Using a single version pattern seems a bit cleaner indeed, will merge it once the CI passes